python/pylib/pose.py

```diff
@@ -20,912 +16,1317 @@ def quatInit():
-def vectorRotateInertialToBody( vIn, eRot ):
-    vResult = np.zeros(np.shape(vIn))
+# Rotate theta eulers from body to inertial frame by ins eulers, in order: phi, theta, psi
+# TODO: this function needs a better description
+def eulerRotateBodyToInertial(e, rot):
+    qe = euler2quat(e)
+    qr = euler2quat(rot)
+    q = mul_Quat_Quat(qr, qe)
+    eul = quat2euler(q)
+    return eul
 
-    for i in range(0,np.shape(vResult)[0]):
-        # Create DCM (rotation matrix)
-        DCM = eulerDCM(eRot[i,:])
-        # Apply rotation to vector
-        vResult[i,:] = np.dot( DCM, vIn[i,:] )
-    
-    return vResult
 
-# Rotate vector from body to inertial frame by euler angles, in order: phi, theta, psi
-def vectorRotateBodyToInertial( vIn, eRot ):
-    vResult = np.zeros(np.shape(vIn))
+# Rotate theta eulers from inertial to body frame by ins eulers, in order: psi, theta, phi
+# TODO: this function needs a better description
+def eulerRotateInertialToBody(e, rot):
+    qe = euler2quat(e)
+    qr = euler2quat(rot)
+    q = mul_ConjQuat_Quat(qr, qe)
+    eul = quat2euler(q)
+    return eul
 
-    for i in range(0,np.shape(vResult)[0]):
-        # Create DCM (rotation matrix)
-        DCM = eulerDCM(eRot[i,:])
-        # Apply rotation to vector
-        vResult[i,:] = np.dot( DCM.T, vIn[i,:] )
-    
-    return vResult
 
 # Rotate vector from inertial to body frame by euler angles, in order: psi, theta, phi
-def vectorRotateInertialToBody2( vIn, eRot ):
-    vResult = np.zeros(np.shape(vIn))
-
-    # Create DCM (rotation matrix)
-    DCM = eulerDCM(eRot)
+# Equivalent to a DCM * vector multiply.
+# TODO: this function needs a better name (see quatConjRot() that does the same but uses quaternions)
+def vectorRotateInertialToBody(euler, v_in):
+    q  = euler2quat(euler)
+    v_out = quatConjRot(q, v_in)
+    return v_out
 
-    for i in range(0,np.shape(vResult)[0]):
-        # Apply rotation to vector
-        vResult[i,:] = np.dot( DCM, vIn[i,:] )
-    
-    return vResult
 
 # Rotate vector from body to inertial frame by euler angles, in order: phi, theta, psi
-def vectorRotateBodyToInertial2( vIn, eRot ):
-    vResult = np.zeros(np.shape(vIn))
-
-    # Create DCM (rotation matrix)
-    DCM = eulerDCM(eRot)
-
-    for i in range(0,np.shape(vResult)[0]):
-        # Apply rotation to vector
-        vResult[i,:] = np.dot( DCM.T, vIn[i,:] )
-    
-    return vResult
-
-
-# Find euler angles of a vector (no roll)
-def vectorEuler( v ):
-    psi     = np.arctan2( v[1], v[0] )
-    dcm     = psiDCM( psi )
-    v2      = np.dot( dcm, v[0:2] )
-    theta   = np.arctan2( -v[2], v2[0] )
-    result  = r_[ 0., theta, psi ]
-#     print("v:", v, " v2:", v2, " e:", result * 180/pi)
-    return result
-
-
-def vectorQuat( v ):
-    return euler2quat( vectorEuler(v) )
+# Equivalent to a DCM.T() * vector multiply.
+# TODO: this function needs a better name (see quatConjRot() that does the same but uses quaternions)
+def vectorRotateBodyToInertial(euler, v_in):
```

Comment:
This function could benefit from a better name. Could be made consistent with quatRot() that does the same but uses quaternions instead of Euler angles